massa-bootstrap/src/establisher.rs

```diff
@@ -44,19 +54,48 @@ impl DefaultListener {
 
         // Number of connections to queue, set to the hardcoded value used by tokio
         socket.listen(1024)?;
-        socket.set_nonblocking(false)?;
-        Ok(DefaultListener(socket.into()))
+        socket.set_nonblocking(true)?;
+        let std_sock = socket.into();
+        let mut mio_listener = mio::net::TcpListener::from_std(std_sock);
```

Comment:
`mio::net::TcpListener::from_std` is needed to have `std::net::TcpStream`, if not used we should use `mio::net::TcpListener::bind(...)`

Reply:
Agreed. If we can go straight to a mio stream, instead of using `from_std`, that would be ideal. I'm just not so sure how to do it, given the issues described in the comment on line 40-42.